ouroboros-network/src/Ouroboros/Network/PeerSelection/State/EstablishedPeers.hs

```diff
@@ -97,10 +97,11 @@ invariant :: Ord peeraddr
           => EstablishedPeers peeraddr peerconn
           -> Bool
 invariant EstablishedPeers {..} =
-       -- The combo of the peer share set + psq = the whole set of peers
-       availableForPeerShare
+       -- The combo of the peer share set + psq = a subset of all peers
+ (  availableForPeerShare
     <> Set.fromList (PSQ.keys nextPeerShareTimes)
-    == Map.keysSet allPeers
+ )
+    `Set.isSubsetOf` Map.keysSet allPeers
```

Comment:
I don't like the fact that this was weakened. Why do we need it?

Reply:
It is needed because `allPeers` contains both peers that support and doesn't support peersharing, but `availableForPeerShare` and `nextPeerShareTimes` only contains peers that support peersharing.

Reply:
I would add a clause to the invariant that checks that all peers in the `availableForPeerShare` set have peer sharing enabled then.

Reply:
I've added a check which makes sure that only peers which support peersharing are in `availableForPeerShare <> nextPeerShareTimes` .